Sources/AsyncQueue/AsyncQueue.swift

```diff
@@ -20,4 +20,70 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
-final class AsyncQueue {}
+/// A queue that enables ordered sending of events from synchronous to asynchronous code
+public final class AsyncQueue: Sendable {
+
+    // MARK: Initialization
+
+    /// Instantiates an asynchronous queue.
+    /// - Parameter priority: The priority of the tasks added to the asynchronous queue.
+    public init(priority: TaskPriority = .medium) {
+        var capturedTaskStreamContinuation: AsyncStream<@Sendable () async -> Void>.Continuation? = nil
+        let taskStream = AsyncStream<@Sendable () async -> Void> { continuation in
+            capturedTaskStreamContinuation = continuation
+        }
+        taskStreamContinuation = capturedTaskStreamContinuation
```

Comment:
```suggestion
        taskStreamContinuation = capturedTaskStreamContinuation!
```
I can't think of any case where this ends up `nil` and we shouldn't blow up anyway.

And deal with any spurious optionality already.

Reply:
This is preferable to making `taskStreamContinuation` implicitly unwrapped because it's more explicit? I went back and forth on this one.

I do agree that at this line the `capturedTaskStreamContinuation` should exist, and it'd be reasonable to blow up the world if it didn't.

Reply:
I mean we can do a guard statement or something cleaner than just `!`, what I prefer is to be done with this optionality as soon as possible since it is an artifact of the fact that the Swift compiler can't tell the value will always be set, not of the logic.